drivers/modem/modem_cellular.c

```diff
@@ -83,11 +84,16 @@ struct modem_cellular_data {
 	uint8_t *chat_argv[32];
 
 	/* Status */
-	uint8_t imei[15];
-	uint8_t hwinfo[64];
 	uint8_t registration_status_gsm;
 	uint8_t registration_status_gprs;
 	uint8_t registration_status_lte;
+	int8_t rssi;
+	uint8_t imei[15];
+	uint8_t model_id[64];
+	uint8_t imsi[15];
+	uint8_t iccid[22];
+	uint8_t manufacturer[64];
+	uint8_t fw_version[12];
```

Comment:
I would say that fw_version is too short. My BG95 for example reports "BG95M3LAR02A03", so 14 characters, and this is a very early fw version. I know that the newer ones have even more characters.